src/extensions/default/WebPlatformDocs/main.js

```diff
@@ -29,98 +29,135 @@ define(function (require, exports, module) {
+        HTMLUtils       = brackets.getModule("language/HTMLUtils"),
+        ExtensionUtils  = brackets.getModule("utils/ExtensionUtils");
+
     // Extension modules
     var InlineDocsViewer = require("InlineDocsViewer");
-    
-    
-    /** @type {?$.Promise} */
-    var _cssDocsPromise = null;
-    
+
+
+    /*
+     * Caches docs promises
+     */
+    var promiseCache = {};
+
     /**
      * Lazily loads JSON docs files. Returns a Promise the is resolved with the parsed Object, or
      * rejected if the file is missing/corrupt.
+     * @param {string} fileName JSON file to load
      * @return {!$.Promise}
      */
-    function getCSSDocs() {
-        if (!_cssDocsPromise) {
+    function getDocs(fileName) {
+        if (!promiseCache[fileName]) {
             var result = new $.Deferred();
-            
-            var path = ExtensionUtils.getModulePath(module, "css.json"),
+            var path = ExtensionUtils.getModulePath(module, fileName),
                 file = FileSystem.getFileForPath(path);
-            
+
             FileUtils.readAsText(file)
                 .done(function (text) {
                     var jsonData;
                     try {
                         jsonData = JSON.parse(text);
                     } catch (ex) {
-                        console.error("Malformed CSS documentation database: ", ex);
+                        console.error("Malformed documentation database: ", path, ex);
                         result.reject();
                     }
                     result.resolve(jsonData);  // ignored if we already reject()ed above
                 })
                 .fail(function (err) {
-                    console.error("Unable to load CSS documentation database: ", err);
+                    console.error("Unable to load documentation database: ", path,  err);
                     result.reject();
                 });
-            
-            _cssDocsPromise = result.promise();
+
+            promiseCache[fileName] = result.promise();
         }
-        
-        return _cssDocsPromise;
+
+        return promiseCache[fileName];
     }
-    
-    
+
+
     /**
-     * Inline docs provider. Currently looks up docs based on CSS properties only.
+     * Inline docs provider.
      *
      * @param {!Editor} editor
      * @param {!{line:Number, ch:Number}} pos
      * @return {?$.Promise} resolved with an InlineWidget; null if we're not going to provide anything
      */
     function inlineProvider(hostEditor, pos) {
-        var langId = hostEditor.getLanguageForSelection().getId();
-        // Only provide docs when cursor is in CSS content
-        if (langId !== "css" && langId !== "scss" && langId !== "less") {
+        var jsonFile, propInfo,
+            propQueue = [], // priority queue of propNames to try
+            langId = hostEditor.getLanguageForSelection().getId(),
+            supportedLangs = ["css", "scss", "less", "html"],
```

Comment:
Can you create separate variable for cssLike language and html language like cssSupportedLangs = ["css", "scss", "less"] and htmlSupportedLangs = ["html"]